catroid/src/main/java/org/catrobat/catroid/content/EventWrapperListener.java

```diff
@@ -48,20 +48,20 @@ public boolean handle(Event event) {
 	}
 
 	private void handleEvent(EventWrapper event) {
-		Collection<EventSequenceAction> sequenceActions = look.sprite.getIdToEventSequenceMap().get(event.eventId);
-		for (EventSequenceAction actionToBeStarted : sequenceActions) {
+		Collection<EventThread> threads = look.sprite.getIdToEventThreadMap().get(event.eventId);
+		for (EventThread threadToBeStarted : threads) {
 			if (event.waitMode == EventWrapper.WAIT) {
-				actionToBeStarted = createActionForWaitEvent(event, actionToBeStarted);
+				threadToBeStarted = createActionForWaitEvent(event, threadToBeStarted);
 			}
-			look.stopActionWithScript(actionToBeStarted.getScript());
-			look.startAction(actionToBeStarted);
+			look.stopThreadWithScript(threadToBeStarted.getScript());
+			look.startThread(threadToBeStarted);
 		}
 	}
 
-	private EventSequenceAction createActionForWaitEvent(EventWrapper event, EventSequenceAction actionToBeStarted) {
+	private EventThread createActionForWaitEvent(EventWrapper event, EventThread threadToBeStarted) {
 		event.addSpriteToWaitFor(look.sprite);
-		actionToBeStarted = actionToBeStarted.clone(); // we want to add an action, but not to the original action
-		actionToBeStarted.setNotifyAction((NotifyEventWaiterAction) ActionFactory.createNotifyEventWaiterAction(look.sprite, event));
-		return actionToBeStarted;
+		threadToBeStarted = threadToBeStarted.clone(); // we want to add an action, but not to the original action
```

Comment:
Please remove the comment if it is no longer accurate. Also I would expect createActionFromWaitEvent(...) to return an action instead of an EventThread. And finally couldn't EventThread have a copy constructor that handles this?